webrecorder/browsertrix
# Add initial crawl template form

WIP https://github.com/ikreymer/browsertrix-cloud/issues/74:
- New list route at `/archives/:id/crawl-templates` with basic list of crawl IDs
- New create route at `/archives/:id/crawl-templates/new`
- Initial form

TODOs (will be addressed in future PRs):
- cron schedule time of day
- descriptions of each field
- advanced settings
- user feedback on success & error
- limit validation

### Manual testing
1. Run app and log in, go to http://localhost:9870/archives
2. Click "Crawl Templates"
  - Verify you're redirected to `/new` if you don't have any crawl templates
  - Verify that you see a list of template IDs if you do have crawl templates
  - Click "Create new crawl template" if you do have templates. Verify it takes you to `/new`
3. Interact with form and submit. Verify you're redirected to crawl templates page and crawl is saved as expected.

Submitted JSON should look something like this:
```json
{
  "name": "Example crawl 1642367874045",
  "schedule": "@weekly",
  "runNow": true,
  "crawlTimeout": 60,
  "config": {
    "seeds": [
      "https://webrecorder.net"
    ],
    "scopeType": "prefix",
    "limit": 1
  }
}
```

### Screenshots
<img width="676" alt="Screen Shot 2022-01-16 at 1 21 14 PM" src="https://user-images.githubusercontent.com/4672952/149678546-68080d08-9525-4c1a-9f8d-9f4fde5b4e26.png">

Schedule options:
<img width="93" alt="Screen Shot 2022-01-16 at 1 21 21 PM" src="https://user-images.githubusercontent.com/4672952/149678549-5d872f5a-ca3d-48ee-84ad-a5596f2c20bd.png">

Scope type options:
<img width="477" alt="Screen Shot 2022-01-16 at 1 21 31 PM" src="https://user-images.githubusercontent.com/4672952/149678560-afc7df0c-ea21-48a3-bd73-005de502582a.png">
